lib/watir-webdriver/elements/checkbox.rb

```diff
@@ -17,14 +17,7 @@ class CheckBox < Input
     #
 
     def set(bool = true)
-      assert_exists
-      assert_enabled
-
-      if @element.selected?
-        @element.click unless bool
-      else
-        @element.click if bool
-      end
+      set? == bool ? assert_enabled : click
```

Comment:
I'm not sure I understand this change correctly.

1. When checkbox is set and user tries to set it - `#assert_enabled` is called.
2. When checkbox is set and user tries to unset it - `#assert_enabled` is not called.

Since previously `#assert_enabled` was always called, I believe this logic is not correct.